### PR TITLE
[TTAHUB-11] Add AR Next Step Complete Date

### PR DESCRIPTION
**Pull request by AdamAdHocTeam**
## Description of change

Add a complete date to next steps.

This is a copy of PR #909 that can be merged into the **main-ar-redesign** branch when ready.

## How to test

Create a new report. On the 'Next Steps' page Create / Add / Remove / Update new steps.

Ultimately you should be able to create and submit and approve a report containing Next Steps with dates.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-11

### Every PR

- [x] Meets issue criteria
- [x] JIRA ticket status updated
- [x] Code is meaningfully tested
- [x] Meets accessibility standards (WCAG 2.1 Levels A, AA)
~- [ ] API Documentation updated~
~- [ ] Boundary diagram updated~
~- [ ] Logical Data Model updated~
~- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions~

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [x] Update JIRA ticket status

### CODE REVIEW

**Comment by thewatermethod** [comment]
Do we want it to validate strings on blur? I entered some nonsense into the date picker text box- "dfg" - and got no validation errors.

Also, should the red error border be as tall as all the fields, or just the next step that's invalid?
See screenshot. I also notice that I have a valid date there with a red border around the box.

Screen Shot 2022-06-29 at 3 18 49 PM

File: frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.css [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
 /* Outline a blank step onBlur or form Validation */
-.ttahub-next-steps-repeater .usa-form-group--error .blank-next-step textarea,
-.usa-form-group--error .ttahub-next-steps-repeater .blank-next-step textarea {
-    border: 3px solid #b50909;
-  }
\ No newline at end of file
+.ttahub-next-steps-repeater .usa-form-group--error .blank-next-step textarea ,
+.usa-form-group--error .ttahub-next-steps-repeater .blank-next-step textarea
+{
+    border: 2px solid #b50909;
```
Comment (thewatermethod) [inline frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.css:10]: Can you convert this to a .scss file and use the brand color explicitly? Check out any other scss file for the syntax

File: frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js [hunk 2]
```diff
@@ -76,68 +97,112 @@ export default function NextStepsRepeater({
...
+                <span className="smart-hub--form-required font-family-sans font-ui-xs text-secondary-dark">
+                  {' '}
+                  *
+                </span>
+              </Label>
+              {blurDateValidations[index]
+                || (errors[name] && errors[name][index]
+                  && errors[name][index].completeDate)
+                ? <ErrorMessage>Enter a complete date</ErrorMessage>
+                : null}
```
Comment (thewatermethod) [inline frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js:182]: I would double check on the message here. I would think we'd want it to say "Enter a valid date" like the other date pickers.

File: frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js [hunk 3]
```diff
@@ -76,68 +97,112 @@ export default function NextStepsRepeater({
...
+              >
+                <Textarea
+                  key={item.key}
+                  id={`${stepType}-next-step-${index + 1}`}
+                  className="height-10 minh-5 smart-hub--text-area__resize-vertical"
+                  name={`${name}[${index}].note`}
+                  type="text"
+                  defaultValue={item.note}
+                  inputRef={register({ required: 'Enter a next step' })}
+                  onBlur={({ target: { value } }) => validateStepOnBlur(value, index)}
```
Comment (thewatermethod) [inline frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js:140]: Maybe you already did, but I would check this validation message against the mock up as well.

File: frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.css (hunk 1 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.css]: Updated to be a SCSS file and use the pre defined colors.

File: frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js (hunk 2 above)
Comment (AdamAdHocTeam) [reply frontend/src/pages/ActivityReport/Pages/components/NextStepsRepeater.js:182]: Updated error message to read "Enter a valid  date".

**Review by thewatermethod — APPROVED** [review]

**Pull request merged**